keps/sig-api-machinery/2339-storageversion-api-for-ha-api-servers/README.md

```diff
@@ -347,6 +349,39 @@ correct order.
 [enables]:https://github.com/kubernetes/kubernetes/blob/220498b83af8b5cbf8c1c1a012b64c956d3ebf9b/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L703
 [filter]:#updating-storageversion
 
+#### Agreements
+
+1. Storageversion updates will be triggered in the following scenarios:
+   1. when a CRD create or update request is received - to reconcile storageversion with the requested CRD changes
+   2. when the apiserver is started - to ensure that we always reconcile storageversions with the current state of CRDs (in case of server restarts)
+2. We will perform async updates of storageversions such that we do not block writes of other unrelated APIs(for ex, crdA writes waiting for update of crdB's storageversion). These requests should be handled asynchronously
+3. We will block a storageversion update until the teardown of prev CRD storage is finished. If reconciliation finds storageversions requiring update and it detects that the CRD storage has already been updated, the storageversion can be updated immediately
+   1. this means we will wait to update storageversion of a CRD until
+      1. old storage of the same CRD is deleted
+      2. pending CR writes using old storageversion of the same CRD are completed
+   2. We will do this to prevent publishing a newer storageversion while a pending CR write is still in progress. Otherwise, if the pending CR write finishes writing the CR using an old version, the storageversion API would not reflect that and the object will remain in etcd, encoded in an old version forever 
+4. We will block new CR writes for a CRD until we have published its latest storageversion. This is discussed more in the [limitations] section
```

Comment:
Curious about the meaning of "published" in this context. Does this imply that any future reads of the StorageVersion resource are guaranteed to reflect the update?

Reply:
> Does this imply that any future reads of the StorageVersion resource are guaranteed to reflect the update?

yes, this was the intention for this statement. That once a CRD update happens, and its storageversion has been updated(or published to the SV API) only then will we unblock the CR writes.